test/test_scan.cpp

```diff
@@ -175,4 +180,164 @@ boost::compute::exclusive_scan(
     CHECK_RANGE_EQUAL(int, 4, output, (0, 1, 3, 6));
 }
 
+BOOST_AUTO_TEST_CASE(inclusive_scan_int_multiplies)
+{
+//! [inclusive_scan_int_multiplies]
+// setup input
+int data[] = { 1, 2, 1, 2, 3 };
+boost::compute::vector<int> input(data, data + 5, queue);
+
+// setup output
+boost::compute::vector<int> output(5, context);
+
+// inclusive scan with multiplication
+boost::compute::inclusive_scan(
+    input.begin(), input.end(), output.begin(),
+    boost::compute::multiplies<int>(), queue
+);
+
+// output = [1, 2, 2, 4, 12]
+//! [inclusive_scan_int_multiplies]
+
+    BOOST_CHECK_EQUAL(input.size(), size_t(5));
+    BOOST_CHECK_EQUAL(output.size(), size_t(5));
+
+    CHECK_RANGE_EQUAL(int, 5, output, (1, 2, 2, 4, 12));
+
+    // in-place inclusive scan
+    CHECK_RANGE_EQUAL(int, 5, input, (1, 2, 1, 2, 3));
+    boost::compute::inclusive_scan(input.begin(), input.end(), input.begin(),
+                                   boost::compute::multiplies<int>(), queue);
+    CHECK_RANGE_EQUAL(int, 5, input, (1, 2, 2, 4, 12));
+}
+
+BOOST_AUTO_TEST_CASE(exclusive_scan_int_multiplies)
+{
+//! [exclusive_scan_int_multiplies]
+// setup input
+int data[] = { 1, 2, 1, 2, 3 };
+boost::compute::vector<int> input(data, data + 5, queue);
+
+// setup output
+boost::compute::vector<int> output(5, context);
+
+// exclusive_scan with multiplication
+// initial value equals 10
+boost::compute::exclusive_scan(
+    input.begin(), input.end(), output.begin(),
+    int(10), boost::compute::multiplies<int>(), queue
+);
+
+// output = [10, 10, 20, 20, 40]
+//! [exclusive_scan_int_multiplies]
+
+    BOOST_CHECK_EQUAL(input.size(), size_t(5));
+    BOOST_CHECK_EQUAL(output.size(), size_t(5));
+
+    CHECK_RANGE_EQUAL(int, 5, output, (10, 10, 20, 20, 40));
+
+    // in-place exclusive scan
+    CHECK_RANGE_EQUAL(int, 5, input, (1, 2, 1, 2, 3));
+    bc::exclusive_scan(input.begin(), input.end(), input.begin(),
+                       int(10), bc::multiplies<int>(), queue);
+    CHECK_RANGE_EQUAL(int, 5, input, (10, 10, 20, 20, 40));
+}
+
+BOOST_AUTO_TEST_CASE(inclusive_scan_int_multiplies_long_vector)
+{
+    size_t size = 1000;
+    bc::vector<int> device_vector(size, int(2), queue);
+    BOOST_CHECK_EQUAL(device_vector.size(), size);
+    bc::inclusive_scan(device_vector.begin(), device_vector.end(),
+                       device_vector.begin(), bc::multiplies<int>(), queue);
+
+    std::vector<int> host_vector(size, 2);
+    BOOST_CHECK_EQUAL(host_vector.size(), size);
+    bc::copy(device_vector.begin(), device_vector.end(), host_vector.begin());
```

Comment:
You should use `queue` here too (and for the rest of the algorithm calls in this test). In general, the tests should always use the context and command queue set up in `context_setup.hpp` instead of falling back to the default queue.